From 4c9133332fa8949dbac539a66577d59b4deb5d87 Mon Sep 17 00:00:00 2001
From: Yu Kuai <yukuai3@huawei.com>
Date: Mon, 11 Jul 2022 09:20:13 +0800
Subject: [PATCH 3503/4585] [Huawei] Revert "[Huawei] mm/filemap: fix that
 first page is not mark accessed in filemap_read()"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 186896

--------------------------------

This reverts commit 22655bbc14858ee3674c0fcb644214b06289f719, prepare
to backport solution from mainline.

Signed-off-by: Yu Kuai <yukuai3@huawei.com>
---
 mm/filemap.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/mm/filemap.c b/mm/filemap.c
index ebae261f9df9..9e209e8a3b0d 100644
--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -2551,11 +2551,10 @@ ssize_t generic_file_buffered_read(struct kiocb *iocb,
 				flush_dcache_page(pages[i]);
 
 			copied = copy_page_to_iter(pages[i], offset, bytes, iter);
-			if (copied) {
-				ra->prev_pos = iocb->ki_pos;
-				written += copied;
-				iocb->ki_pos += copied;
-			}
+
+			written += copied;
+			iocb->ki_pos += copied;
+			ra->prev_pos = iocb->ki_pos;
 
 			if (copied < bytes) {
 				error = -EFAULT;
-- 
2.17.1

